frontend/src/metabase/dashboard/components/DashCard.jsx

```diff
@@ -323,6 +323,7 @@ const DashCardActionButtons = ({
   if (getVisualizationRaw(series).visualization.supportPreviewing) {
     buttons.push(
       <ToggleCardPreviewButton
+        key="toggle-card-preview-button"
```

Comment:
This is very good finding! Unique key for component is always important to avoid surprises.
Can you perhaps split the changes of this `DashCard.jsx` into its own PR so we can get it merged separately.

Reply:
Preparing this PR for this exact reason: grouping all the findings discovered while investigating #12985.